projecttacoma/deqm-test-server-frontend
# New resource

## Summary

“Create new [resourceType]” button added to each resource type page, which links to a new page with a  code editor that enables the user to type in a new JSON resource. Submit button is only clickable if the code is valid JSON. 

## New behavior

"Create new [resourceType]" button links to a new page with a code editor allowing a user to enter a new resource. 

## Code changes

Components/ResourceCodeEditor: created a code editor component where the user can type in a JSON resource and a submit button that only works if the code is valid JSON
Components/Create.tsx: created a page that displays a ResourceCodeEditor component 
Components/Index.tsx: added “create new [resourceType]” button linking to Create.tsx
Unit tests added to __tests__ to test the above
A few files contain UI updates, and a few mantine installations were added to package.json and package-lock.json

## Testing guidance

- Have a .env.local file and add the following to it: NEXT_PUBLIC_DEQM_SERVER=https://abacus-dev.mitre.org/4_0_1 However, to run a local version of the test server, change this NEXT_PUBLIC_DEQM_SERVER to the desired port (this project runs on the default 3000).
- Run the frontend w npm run dev, then navigate to http://localhost:3000/. Click on any resource, then click on the “create new [resourceType]” button. Type invalid JSON code to verify the submit button doesn’t work. Then type valid JSON code, click the submit button, and verify that your code console logs. 
- navigate to http://localhost:3000/[resourceType]/ create directly and repeat the steps above.
